File: csharp/LockController/LockControllerTest/LockControllerTest.cs
using LockControllerLib;
using Xunit;

namespace LockControllerTest;

public class LockControllerTest
{
    [Fact]
    public void LockThenUnlock()
    {
        // TODO: finish writing this test
        var m = new MassiveObject("Id1");
        IMassiveDbConnection connection = null; // TODO
        var controller = new LockController(connection);
        Assert.True(controller.IsSubscriptionActive());
        controller.LockObject(m.Id);
        Assert.True(controller.IsLocked(m.Id));
        controller.UnlockObject(m.Id);
        Assert.False(controller.IsLocked(m.Id));
        controller.Dispose();
        Assert.False(controller.IsSubscriptionActive());
    }
}

File: typescript/tests/LockController.test.ts
import { IMassiveDbConnection } from '../src/IMassiveDbConnection';
import { MassiveObject } from '../src/IMassiveObject';
import { LockController } from '../src/LockController';

describe("LockController", () => {
  it("locks and then unlocks", () => {
    // TODO: finish writing this test
    const m = new MassiveObject("Id1");
    const connection: IMassiveDbConnection = null as any; // TODO
    const controller = new LockController(connection);
    expect(controller.isSubscriptionActive()).toBeTruthy();
    controller.lockObject(m.Id);
    expect(controller.isLocked(m.Id)).toBeTruthy();
    controller.unlockObject(m.Id);
    expect(controller.isLocked(m.Id)).toBeFalsy();
    controller.dispose();
    expect(controller.isSubscriptionActive()).toBeFalsy();
  })
})
